packages/@o3r/rules-engine/src/engine/ruleset-executor.ts

```diff
@@ -195,38 +196,65 @@ export class RulesetExecutor {
     throw new Error(`Unknown condition block met : ${JSON.stringify(nestedCondition)}`);
   }
 
+  // TODO: https://github.com/AmadeusITGroup/otter/issues/1044
   /**
    * Plug ruleset to fact streams and trigger a first evaluation
-   *
-   * @param ruleset
    */
   public prepareRuleset() {
-    const rulesWithContext: Rule[] = [];
-    const rulesWithoutContext: Rule[] = [];
+    const inputFactsForRule: Record<string, string[]> = {};
+    /**
+     * Recursively explores a rule to identify and collect input facts.
+     * Input facts are identified based on the 'FACT' type and operator-specific implicit dependencies.
+     *
+     * @param currentObject The current object being explored.
+     * @param ruleInputFacts A set to store the identified input facts for the rule.
+     */
+    const collectRuleInputFacts = (currentObject: any, ruleInputFacts: Set<string>) => {
```

Comment:
`currentObject` should be properly typed.
(this function can also be a private field to not be redefined at each execution)